students/alex_skrn/Lesson06Assignment/water-regulation/water-regulation/waterregulation/test.py

```diff
@@ -0,0 +1,221 @@
+"""Unit tests for the water-regulation module."""
+
+import unittest
+from unittest.mock import MagicMock
+
+from pump import Pump
+from sensor import Sensor
+
+from .controller import Controller
+from .decider import Decider
+
+
+class DeciderTests(unittest.TestCase):
+    """Unit tests for the Decider class."""
+
+    # # TODO: write a test or tests for each of the behaviors defined for
+    # #       Decider.decide
+    #
+    # def test_dummy(self):
+    #     """
+    #     Just some example syntax that you might use
+    #     """
+    #
+    #     pump = Pump('127.0.0.1', 8000)
+    #     pump.set_state = MagicMock(return_value=True)
+    #
+    #     self.fail("Remove this test.")
+
+    def setUp(self):
+        """Run each time before any test method is run."""
+        self.pump = Pump('127.0.0.1', 8000)
+
+        self.actions = {'PUMP_IN': self.pump.PUMP_IN,
+                        'PUMP_OUT': self.pump.PUMP_OUT,
+                        'PUMP_OFF': self.pump.PUMP_OFF,
+                        }
+
+        self.decider = Decider(100, 0.05)
+
+    def test_decider_init(self):
+        """Test that decider gets intantiated with expected params."""
+        self.assertIsInstance(self.decider, Decider)
+        self.assertEqual(self.decider.target_height, 100)
+        self.assertEqual(self.decider.margin, 0.05)
+
+    def test_off_and_below_then_pump_in(self):
+        """Test behavior 1.
+
+        1. If the pump is off and the height is below the margin region,
+        then the pump should be turned to PUMP_IN.
+        """
+        self.assertEqual(self.decider.decide(90,
```

Comment:
I would suggest instead of cramming this all into the `self.assertEqual` statement that you create an `expected` and `actual` variables separately and then compare.  I think it looks a little cleaner